cdap-proto/src/main/java/co/cask/cdap/proto/codec/AuditMessageTypeAdapter.java

```diff
@@ -39,8 +40,13 @@ public AuditMessage deserialize(JsonElement json, Type typeOfT, JsonDeserializat
     throws JsonParseException {
     JsonObject jsonObj = json.getAsJsonObject();
     long timeMillis = jsonObj.get("time").getAsLong();
-    MetadataEntity metadataEntity = context.deserialize(jsonObj.getAsJsonObject("metadataEntity"),
-                                                        MetadataEntity.class);
+    MetadataEntity metadataEntity;
+    EntityId entityId = context.deserialize(jsonObj.getAsJsonObject("entityId"), EntityId.class);
+    if (entityId != null) {
```

Comment:
After our discussed change, this will also not be required. Please make a note to remove this then.

Reply:
Yes this will be changed back in the backward compatibility change.